moulinette/interfaces/cli.py

```diff
@@ -490,7 +490,11 @@ def _do_authenticate(self, authenticator):
         Handle the core.MoulinetteSignals.authenticate signal.
 
         """
-        # TODO: Allow token authentication?
+        # Hmpf we have no-use case in yunohost anymore where we need to auth
+        # because everything is run as root ...
+        # I guess we could imagine some yunohost-independant use-case where
+        # moulinette is used to create a CLI for non-root user that needs to
+        # auth somehow but hmpf -.-
         help = authenticator.extra.get("help")
         msg = m18n.n(help) if help else m18n.g("password")
         return authenticator(password=self._do_prompt(msg, True, False, color="yellow"))
```

Comment:
As noted, not sure what to do here ... I was thinking maybe to keep a mechanism using auth through an env variable .. I think that's similar to something I saw for postgresql ... So like 

```
MOULINETTE_PASSWORD="foobar" yunohost some command
```

(though in the context of yunohost we have no need for this, but if we consider moulinette as an independent piece of software it could be useful for some use case) ... I'm not even sure why I wanted to get rid of this "interactive auth" mechanism ... Maybe to make it independent of os.isatty earlier ... but maybe we don't care at all and the current code is fine ...

Reply:
You clearly have a better view of this project than I do, but we're supposed to handle multiple authenticators:
https://github.com/YunoHost/moulinette/blob/d2cb8027927002e06e4309963d593ad3d853090f/test/actionsmap/moulitest.yml#L9-L22

And the tests I've written show that it works as expected. I'm not sure it would be easy to keep this potential feature if we chose to store the password in one file (maybe with different files? but what about ldap authenticator?).

Reply:
Eh not sure to understand where I was implying a password stored in a file, I was meaning that instead of asking the password interactively it could be provided through an env variable ... But anyway I think we can just keep the current code and it's fine, sorry, that's probably just my talent about overthinking everything :sweat_smile:

Reply:
Yeah me neither, sorry I'm doing 100 things all together :S

Env var? that's a solution indeed :smile: